files/shaders/objects_fragment.glsl

```diff
@@ -146,9 +147,9 @@ void main()
     float shininess = gl_FrontMaterial.shininess;
     vec3 matSpec = gl_FrontMaterial.specular.xyz;
 #endif
-
     gl_FragData[0].xyz += getSpecular(normalize(viewNormal), normalize(passViewPos.xyz), shininess, matSpec);
 
+    gl_FragData[0].xyz = mix(ambientBias.y * gl_FragData[0].xyz, gl_FragData[0].xyz, shadowing);
```

Comment:
This isn't a good way of doing things. Applying the shadow to just the light casting shadows is a better idea.

Reply:
I say it : simplistic I let you do what u want as i don't master the 2 passes terrain shading

Reply:
This has nothing to do with terrain shading at all. I can't tell what you're trying to say here.

Reply:
so tell me where to put the line